src/libs/antares/study/study.h

```diff
@@ -449,6 +449,12 @@ class Study: public Yuni::NonCopyable<Study>, public IObject, public LayerData
     void removeTimeseriesIfTSGeneratorEnabled();
     //@}
 
+    /*!
```

Comment:
The idea is to first read currentStudyVerstion and save it value to temporary variable, after that we are calling `study->saveToFolder`. Why we need temporary variable? We need it because, if we are upgrading a study, variable `study->.header.version` will be changed to a current version of our software, and if we are passing value of the `study->.header.version` variable to the `deleteDeprecatedHydroMaxPowerTSFiles` function we will be passing already value of our study version (currently 870). If we are upgrading for example from 8.2 version than we need to pass 820 value to the `deleteDeprecatedHydroMaxPowerTSFiles` function, and we are doing that via temporary variable.
Condition `if (currentStudyVersion >= 870)`  must be changed to `if(temporaryVersionVar < 870)` . Why? Because  currentStudyVersion after saving a study will have 870 value, and deletion won't be performed.

Reply:
Also, we need to delete files just if `study->saveToFolder` return true, than we are sure that our new files are saved and all data from deprecated files are stored in our new files.